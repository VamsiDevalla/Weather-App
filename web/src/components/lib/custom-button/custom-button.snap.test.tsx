import React from 'react';
import { create } from 'react-test-renderer';
import CustomButton from './custom-button.component';

describe('Custom Button', () => {
  test('should render properly for button type submit', () => {
    expect(
      create(
        <CustomButton id='test' type='submit' onClick={jest.fn()}>
          Test
        </CustomButton>,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  test('should render properly for button type button', () => {
    expect(
      create(
        <CustomButton id='test' type='button' onClick={jest.fn()}>
          Test
        </CustomButton>,
      ).toJSON(),
    ).toMatchSnapshot();
  });
  test('should render properly for button type button', () => {
    const clickHandler = jest.fn();
    const button = create(
      <CustomButton id='test' type='button' onClick={clickHandler}>
        Test
      </CustomButton>,
    );
    button.root.findByType('button').props.onClick();
    expect(clickHandler).toHaveBeenCalledTimes(1);
  });
});
