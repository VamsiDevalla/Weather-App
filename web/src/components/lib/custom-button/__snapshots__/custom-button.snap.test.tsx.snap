// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Button should render properly for button type button 1`] = `
<button
  className="custom-button"
  data-testid="button"
  id="test"
  onClick={[MockFunction]}
  type="button"
>
  Test
</button>
`;

exports[`Custom Button should render properly for button type submit 1`] = `
<button
  className="custom-button"
  data-testid="button"
  id="test"
  onClick={[MockFunction]}
  type="submit"
>
  Test
</button>
`;
